File: StoreManagement/src/models/UserRegisterDTO.ts
import { Gender } from "./Employee";
import { MaritalStatus } from "./UserProfile";

// Contains all User fields and all UserProfile fields
export interface UserRegisterDTO {
    name: string;
    password: string;

    bio: string;
    location: string;

    birthday?: string;
    gender: Gender;
    maritalStatus: MaritalStatus;
}

File: StoreAPI/StoreAPI/Models/UserRegisterDTO.cs
namespace StoreAPI.Models
{
    public class UserRegisterDTO
    {
        public virtual string? Name { get; set; }
        public virtual string? Password { get; set; }

        public virtual string? Bio { get; set; }
        public virtual string? Location { get; set; }

        public virtual DateTime? Birthday { get; set; }
        public virtual Gender Gender { get; set; }
        public virtual MaritalStatus MaritalStatus { get; set; }
    }
}
